product_pricelist_fixed_price/model/product_pricelist_item.py

```diff
@@ -1,50 +1,124 @@
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as published
-#    by the Free Software Foundation, either version 3 of the License, or
-#    (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-from openerp import models, fields, api, _
+# © 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com).
+#        Pedro M. Baezr <pedro.baeza@serviciosbaeza.com>.
+# © 2015 Therp BV (http://therp.nl).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models, _
 
 
-class ProductPricelistItem(models.Model):
+FIXED_PRICE_TYPE = -3
+
 
+class ProductPricelistItem(models.Model):
+    """Add fixed price option to pricelist item."""
     _inherit = 'product.pricelist.item'
 
     def _price_field_get_ext(self):
+        """Add fixed price to pricetype selection."""
         result = self._price_field_get()
-        result.append((-3, _('Fixed Price')))
+        result.append((FIXED_PRICE_TYPE, _('Fixed Price')))
         return result
 
-    base_ext = fields.Selection(selection='_price_field_get_ext',
-                                string='Based on',
-                                size=-1,
-                                required=True,
-                                default=lambda self:
-                                    self.default_get(
-                                        fields_list=['base'])['base'],
-                                    help="Base price for computation")
+    base_ext = fields.Selection(
+        selection='_price_field_get_ext',
+        string='Based on',
+        size=-1,  # Needed when selection is for integer based values.
+        required=True,
+        default=lambda self:
+            self.default_get(
+                fields_list=['base']
+            )['base'],
+        help="Base price for computation",
+    )
+
+    @api.constrains('base_ext')
+    def _check_fixed_price(self):
+        """Ensure fixed prices always refer to a specific product.
+
+        Values for price_discount and price_round will not be checked,
+        because create and write will automagically set appropiate
+        values.
+        """
+        if self.base_ext == FIXED_PRICE_TYPE and not self.product_id:
+            raise ValidationError(
+                _("Product required for fixed price item.")
+            )
+
+    def _auto_end(self, cr, context=None):
```

Comment:
If the `post_init_hook()` is really executed (see comment above), this method is (normally) useless